content/articles/getting-started-with-hookstate/index.md

```diff
@@ -0,0 +1,186 @@
+
+React is a mature JavaScript library that we use to create declarative and dynamic UIs. React breaks the UI into components. However, over a large codebase, these components will need to share data between them. For this reason, the concept of state in React development is crucial. We need to understand - what it is, how to properly manage it, and how to handle complexity as the application grows.
+
+### Goal
+In JavaScript web applications, state refers to the object that holds information generated through user actions. Since in modern applications we break the UI into components, these components depend on dynamic data. This article will cover the essentials of state management and the efficiencies that the Hookstate library provides in React applications.
+
+### Prerequisites
+- A web browser such as [Google Chrome](https://www.google.com/chrome/browser-tools/).
+- A code editor such as [VS Code](https://code.visualstudio.com/) or an IDE.
+- Knowledge of the [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript) programming language.
+- Basics of the [React library](https://reactjs.org/).
+- Have a Node.js LTS or later version on your development environment. You can download [here](https://nodejs.org/en/download/).
+
+### Understanding State and Hookstate
+In a nutshell, state management is a pattern in which we control the communication and sharing of data across the reusable components of the modern [frontend frameworks](https://www.sitepoint.com/most-popular-frontend-frameworks-compared/). To achieve this, our application wraps data in a data structure that will represent our application's logical state that we can access and manipulate to transition between these states. Often, the changes are made depending on the user's actions.
+
+A case study can be an e-commerce application. To build such an app, we can break the UI into components such as `Button`, `Cart`, `Checkout`, `Login`, and more. When a user adds an item to the `Cart` component or performs a successful login, these actions will alter the state of the component and hence the entire application. In a large application, it can become tedious work to track the global state while still [prop drilling] is complicated and redundant. This is where state management libraries come in.
+
+Hookstate library takes the management of state in React to a new level. The library wraps the concept of the declarative React `useState` to a global version and extends it with useful features.
```

Comment:
Explain more about HookState in a new paragraph. What are it's features? How is it different from existing solutions? What other solutions do we have? How does it solve the problem?